lib/src/rating/rating.component.ts

```diff
@@ -0,0 +1,231 @@
+import { Component, Input, forwardRef, ViewEncapsulation, ViewChildren, QueryList, ElementRef, OnInit } from "@angular/core";
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from "@angular/forms";
+
+export const CUSTOM_RATING_VALUE_ACCESSOR: any = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => RatingComponent),
+    multi: true,
+};
+
+@Component({
+    selector: "ac-rating",
+    templateUrl: "./rating.component.html",
+    styleUrls: ["./rating.component.scss"],
+    providers: [CUSTOM_RATING_VALUE_ACCESSOR],
+    encapsulation: ViewEncapsulation.None,
+})
+export class RatingComponent implements ControlValueAccessor, OnInit {
+    @Input() iconWidth?: number = 25;
+    @Input() iconHeight?: number = 25;
+    @Input() max?: number = 5;
+    @Input() readOnly?: boolean = false;
+    @Input() tooltipList?: Array<string>;
+    @Input() className?: string;
+    @Input() useHollow?: boolean;
+    @Input() showValue?: boolean;
+    @Input() showTextValue?: boolean;
+
+    randomIds: Array<string> = [];
+
+    /**
+     * @member 0 Grey (unselected)
+     * @member 1 Yellow (Selected)
+     */
+    @Input() colors?: Array<string> = ["#A9A9A9", "#ffc500"];
+
+    // Placeholders for the callbacks which are later provided
+    // by the Control Value Accessor
+    private innerValue: any = "";
+
+    activeList: Array<string> = [];
+    initialized: boolean = false;
+    displayValue?: string;
+
+    tabCounter: number = 0;
+
+    @ViewChildren("radioItemRefs") radios: QueryList<ElementRef>;
+
+    ngOnInit() {
+        if (this.max) {
+            this.randomIds = Array.apply(null, Array(this.max)).map((item: any) => {
+                const identifier = +Math.floor(Math.random() * 100) + new Date().getTime();
+                return "star-" + identifier;
+            });
```

Comment:
can you use `randomId` from [frontend-tools](https://github.com/sebgroup/frontend-tools/wiki/randomId)